Add BasePlugin deprecation notice and porting tips to plugin development guide (#3400)
Kong/docs.konghq.com: app/gateway/2.7.x/plugin-development/custom-logic.md

end
```

> ### Porting from old `BasePlugin` style.

Reviewer: ```suggestion
### Migrating from `BasePlugin` module
```

The `BasePlugin` module is deprecated and will be removed in a future version

> of {{site.ce_product_name}}.  If you have an old plugin that used it, replace

Reviewer: ```suggestion
of {{site.ce_product_name}}.  If you have an old plugin that uses it, replace
```

the initial part:

```lua
--  DEPRECATED --
local BasePlugin = require "kong.plugins.base_plugin"

local CustomHandler = BasePlugin:extend()

CustomHandler.VERSION  = "1.0.0"
CustomHandler.PRIORITY = 10
```

with the current equivalent:
```lua
local CustomHandler = {
  VERSION  = "1.0.0",
  PRIORITY = 10,
}
```

> No need to add a `:new()` method, nor call the `CustomHandler.super.XXX:(self)`

Reviewer: ```suggestion
You don't need to add a `:new()` method or call any of the `CustomHandler.super.XXX:(self)`
```
